src/api/app/models/bs_request_action.rb

```diff
@@ -822,6 +822,23 @@ def uniq_key
     ' '
   end
 
+  def name
+    uniq_key
+  end
+
+  def details
+    package = Package.find_by_project_and_name(source_project, source_package)
+
+    return { comment: 'Commit information is missing' } if package.nil? || source_rev.nil?
+
+    hash = package.commit(source_rev) || package.commit
+
+    { comment: hash['comment']&.truncate(50) || name,
+    time: "#{ApplicationController.helpers.time_ago_in_words(hash['time'].to_i)} ago",
+    user: hash['user'],
+    rev: hash['rev'] }
```

Comment:
I still don't like the mangling of the hash here. This is view code. I think this method should return `nil`, `package.commit(source_rev)` or `package.commit`. And all the string mangling you can do in the view/component.

Reply:
done

Reply:
:+1: